tests/check_linux_layout_intel.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# Copyright (c) 2024, Intel Corporation.
+
+import argparse
+import os
+import pathlib
+
+sof_ri_info = None
+try:
+    import sof_ri_info
+except ImportError:
+    print("Note: not using sof_ri_info")
+    pass
+
+# List of Intel firmware files and their paths the upstream SOF Linux driver
+# expects to find under /lib/firmware/intel. A sof-bin release should have
+# binaries for all these targets.
+fw_list = [
+    "sof/sof-apl.ri",
+    "sof/sof-adl-n.ri",
+    "sof/sof-ehl.ri",
+    "sof/sof-glk.ri",
+    "sof/sof-adl.ri",
+    "sof/sof-rpl.ri",
+    "sof/community/sof-apl.ri",
+    "sof/community/sof-adl-n.ri",
```

Comment:
This is a lot of error-prone duplication. If you insist doing this in Python:

```python

   unsigned_IPC3 = [ 'apl', 'adl', 'ehl', ... ]   

   expected_files = set()

   for p in unsigned_IPC3:
      expected_files |= { f'sof/sof-{p}.ri', f'sof/community/sof-{p}.ri',
                    f'sof/intel-signed/sof-{p}.ri' }


```

Then the `unsigned_IPC3` and other data can become a simple and short configuration file with a list of platforms, similar to `sof/installer/sample-config.mk`